### PR TITLE
[13.0][FIX] company_default_partner_pricelist

### PR DESCRIPTION
**Pull request by SergiCForgeFlow**
This fix manages the company's default pricelist by utilizing a generic property, taking advantage of the fact that Odoo standard prioritizes a generic property over using the domain hook to obtain the default pricelist.

### CODE REVIEW

File: company_default_partner_pricelist/models/res_company.py [hunk 1]
```diff
@@ -11,3 +11,48 @@ class ResCompany(models.Model):
...
+        )
+        if ppty:
+            if not default_property_pp:
+                ppty.sudo().unlink()
+            else:
+                ppty.sudo().write(
+                    {"value_reference": "product.pricelist,%s" % default_property_pp}
+                )
+        else:
+            if default_property_pp:
```
Comment (JordiMForgeFlow) [inline company_default_partner_pricelist/models/res_company.py]:
```suggestion
        elif default_property_pp:
```

**Comment by JordiMForgeFlow** [comment]
@SergiCForgeFlow could you squash the commits and remove the '[13.0]' in the commit message?

**Comment by SergiCForgeFlow** [comment]
> @SergiCForgeFlow could you squash the commits and remove the '[13.0]' in the commit message?

Done :+1:

File: company_default_partner_pricelist/migrations/13.0.1.0.1/post-migration.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    company = env["res.company"]
+    ir_property = env["ir.property"]
+    field = env["ir.model.fields"]._get("res.partner", "property_product_pricelist")
+    for record in company.search([]):
+        company_default_pricelist = record.default_property_product_pricelist
```
Comment (JordiMForgeFlow) [inline company_default_partner_pricelist/migrations/13.0.1.0.1/post-migration.py:12]:
```suggestion
        company_default_pricelist = record.default_property_product_pricelist
        if not company_default_pricelist:
            continue
```

File: company_default_partner_pricelist/migrations/13.0.1.0.1/post-migration.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            else:
+                ir_property.sudo().create(
+                    {
+                        "name": "property_product_pricelist",
+                        "value_reference": "product.pricelist,%s"
+                        % company_default_pricelist.id,
+                        "fields_id": field.id,
+                        "company_id": record.id,
+                    }
+                )
```
Comment (JordiMForgeFlow) [inline company_default_partner_pricelist/migrations/13.0.1.0.1/post-migration.py]:
```suggestion
            if ppty:
                res = ppty.value_reference.split(",")
                res_id = int(res[1])
                if res_id != record.default_property_product_pricelist.id:
                    ppty.sudo().write(
                        {
                            "value_reference": "product.pricelist,%s"
                            % company_default_pricelist.id
                        }
                    )
            else:
                ir_property.sudo().create(
                    {
                        "name": "property_product_pricelist",
                        "value_reference": "product.pricelist,%s"
                        % company_default_pricelist.id,
                        "fields_id": field.id,
                        "company_id": record.id,
                    }
                )
```

File: company_default_partner_pricelist/migrations/13.0.1.0.1/post-migration.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    field = env["ir.model.fields"]._get("res.partner", "property_product_pricelist")
+    for record in company.search([]):
+        company_default_pricelist = record.default_property_product_pricelist
+        if not company_default_pricelist:
+            continue
+        ppty = ir_property.search(
+            [
+                ("company_id", "=", record.id),
+                ("fields_id", "=", field.id),
+                ("res_id", "=", False),
```
Comment (JordiMForgeFlow) [inline company_default_partner_pricelist/migrations/13.0.1.0.1/post-migration.py:20]:
```suggestion
                ("res_id", "=", False),
                ("value_reference", "!=", False),
```

File: company_default_partner_pricelist/migrations/13.0.1.0.1/post-migration.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                ("value_reference", "!=", False),
+            ],
+            limit=1,
+        )
+        if ppty:
+            res = ppty.value_reference.split(",")
+            res_id = int(res[1])
+            if res_id != record.default_property_product_pricelist.id:
```
Comment (JordiMForgeFlow) [inline company_default_partner_pricelist/migrations/13.0.1.0.1/post-migration.py]:
```suggestion
            if res_id != company_default_pricelist.id:
```

**Review by JordiMForgeFlow — APPROVED** [review]
👍🏼 fix makes sense as Odoo uses the get_multi method to get matching pricelist (https://github.com/odoo/odoo/blob/abd9e6686bdb1f84d67bd536d9cecda627cd001a/addons/product/models/product_pricelist.py#L322), and get_multi will return the generic property (https://github.com/odoo/odoo/blob/abd9e6686bdb1f84d67bd536d9cecda627cd001a/odoo/addons/base/models/ir_property.py#L213).

File: company_default_partner_pricelist/migrations/13.0.1.0.1/post-migration.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    company = env["res.company"]
+    ir_property = env["ir.property"]
+    field = env["ir.model.fields"]._get("res.partner", "property_product_pricelist")
+    for record in company.search([]):
+        company_default_pricelist = record.default_property_product_pricelist
+        if not company_default_pricelist:
+            continue
+        ppty = ir_property.search(
+            [
+                ("company_id", "=", record.id),
```
Comment (MiquelRForgeFlow) [inline company_default_partner_pricelist/migrations/13.0.1.0.1/post-migration.py:18]:
```suggestion
                ("name", "=", "property_product_pricelist"),
                ("company_id", "=", record.id),
```

File: company_default_partner_pricelist/models/res_company.py [hunk 7]
```diff
@@ -11,3 +11,48 @@ class ResCompany(models.Model):
...
+    def _update_partner_pricelist_generic_property(self):
+        self.ensure_one()
+        default_property_pp = self.default_property_product_pricelist
+        ir_property = self.env["ir.property"]
+        field = self.env["ir.model.fields"]._get(
+            "res.partner", "property_product_pricelist"
+        )
+        ppty = ir_property.sudo().search(
+            [
+                ("company_id", "=", self.id),
```
Comment (MiquelRForgeFlow) [inline company_default_partner_pricelist/models/res_company.py:25]:
```suggestion
                ("name", "=", "property_product_pricelist"),
                ("company_id", "=", self.id),
```

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM

**Pull request merged**